rtlib/include/rt/Data.hpp

```diff
@@ -0,0 +1,71 @@
+//
+// Created by Anthony Tamasi on 2019-10-17.
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <boost/filesystem.hpp>
+#include <opencv2/core.hpp>
+
+#include "rt/ImageTypes.hpp"
+#include "rt/types/ITKMesh.hpp"
+
+namespace et = envitools;
+namespace fs = boost::filesystem;
+
+
+namespace Data {
+    class Data
+    {
+        using Pointer = std::shared_ptr<Data>; //DOES THIS GO HERE?
+
+    public:
+        static Pointer Load(const fs::path& path);
+        virtual cv::Mat getImage(int idx = 0) = 0;
+
+    protected:
+        Data() = default;
+        Data(std::string path) {} //Constructor
+
+    };
+
+
+    class Image : public Data
+    {
+    public:
+        Image(const fs::path& path){ /* Do something to load */}
+        cv::Mat getImage() { return img_; } //Return this image
+    private:
+        cv::Mat img_;
+    };
+
+
+
+    class ENVI : public Data
+    {
+    public:
+        ENVI(const fs::path& path){ /* Do something to load */ }
+        //cv::Mat getImage(int idx = 0) { return bands_[idx]; } //Could use this if we read bands into vector. But would also need functions to load into vector, and access vector.
+        cv::Mat getImage(int idx = 0) {}
+    private:
+        //std::vector<cv::Mat> bands_; //Could use to store bands we have read in memory?
+        et::ENVI envi_;
+    };
+
+
+    class Mesh : public Data
+    {
+    public:
+        Mesh(const fs::path& path){ /* Do something to load */ }
+        cv::Mat getCVImage() { return cvFixedImage_; }
+        rt::Image8UC3::Pointer getFixedImage() { return fixedImage_; }
```

Comment:
Only need functions that return `cv::Mat`. It will make life easier.

Reply:
These are the functions that were already in use in the code. Would you rather I just fold the extra two functions into the function that returns a cv::Mat and turn those two into private helper methods or do do something else? Because I know the cv::Mat returning function uses those two functions to get to a cv::Mat.

Reply:
Does the 
```
/** 3D mesh of the input data */
rt::ITKMesh::Pointer origMesh_;
```

line stay in that file? Or does this get taken out, too? If it does get taken out, it doesn't seem like we would need the getMesh/getNumMeshes virtual functions that were suggested.